qiskit/ignis/calibrations/fitters.py

```diff
@@ -0,0 +1,379 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Fitters for calibration routines
+"""
+
+import numpy as np
+from qiskit import QiskitError
+from ..characterization.fitters import BaseFitter
+
+
+class BaseCalibrationFitter(BaseFitter):
```

Comment:
Eventually all the fitters should be able to accept either data input. That's also why I don't want BasePulseFitter and BaseCircuitFitter

Reply:
I'm not sure this is wise to double up on types of data, single responsibility principle. We should have a *most Base* fitter that is agnostic of data type, and we could have one that chooses what type of fitter to instantiate, but at the end of the day, there should be a fitter that handles a specific sort of data. Its important to avoid things that work like magic (ie. passing any type of data to a fitter and having it automatically do its discrimination behind the scenes), these sorts of APIs end up not being very extensible and so if you want to do anything out of the ordinary it becomes hard to put the building blocks together.